packages/react-components/src/organisms/CheckboxGroup.tsx

```diff
@@ -3,16 +3,21 @@ import { v4 as uuidV4 } from 'uuid';
 
 import { noop } from '../utils';
 import { LabeledCheckbox } from '../molecules';
+import { Caption } from '../atoms';
 
 export interface Option<V extends string> {
   value: V;
   label: string;
   enabled?: boolean;
 }
 
-interface CheckboxGroupProps<V extends string> {
-  readonly options: ReadonlyArray<Option<V>>;
+export interface Title {
+  title: string;
+  label?: undefined;
+}
 
+interface CheckboxGroupProps<V extends string> {
+  readonly options: ReadonlyArray<Option<V> | Title>;
```

Comment:
I think this type is the one that needs to be exported and the other two shouldn't need to be. Then use it where i mark below